controllers/platformnetwork_controller.go

```diff
@@ -289,7 +315,14 @@ func (r *PlatformNetworkReconciler) Reconcile(ctx context.Context, request ctrl.
 
 	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&
 		instance.Status.Reconciled {
-		return ctrl.Result{}, nil
+
+		// Update the deploymentScope if necessary
+		err, scope_updated = r.UpdateDeploymentScope(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		} else if !scope_updated {
+			return ctrl.Result{}, nil
+		}
 	}
 
 	// Update scope from configuration
```

Comment:
Here we update the deploy scope again?

Reply:
Not just scope but we are updating other things such as `ObservedGeneration`, `ConfigurationUpdated`, `ReconcileAfterInSync` annotation and in case of host controller few other status attributes such `HostProfileConfigurationUpdated`.

All above attributes are updated when `Generation != ObservedGeneration` which is different from `UpdateDeploymentScope` which updates only the `deploymentScope` if user wants to do it when `Generation == ObservedGeneration`.